gradio/analytics.py

```diff
@@ -27,13 +41,50 @@ def analytics_enabled() -> bool:
 
 
 def _do_analytics_request(url: str, data: dict[str, Any]) -> None:
+    if wasm_utils.IS_WASM:
+        asyncio.ensure_future(
+            _do_wasm_analytics_request(
+                url=url,
+                data=data,
+            )
+        )
+    else:
+        threading.Thread(
+            target=_do_normal_analytics_request,
+            kwargs={
+                "url": url,
+                "data": data,
+            },
+        ).start()
+
+
+def _do_normal_analytics_request(url: str, data: dict[str, Any]) -> None:
     data["ip_address"] = get_local_ip_address()
     try:
         requests.post(url, data=data, timeout=5)
     except (requests.ConnectionError, requests.exceptions.ReadTimeout):
         pass  # do not push analytics if no network
 
 
+async def _do_wasm_analytics_request(url: str, data: dict[str, Any]) -> None:
+    data["ip_address"] = "No internet connection"
```

Comment:
Since the current `get_local_ip_address()` doesn't work in the Wasm env, and even if we make its Wasm-compatible version, gathering users' public IP-address looks too much, I hardcoded the string literal representing the failure of IP address detection.

Reply:
> gathering users' public IP-address looks too much

What do you mean by looks too much. Do you mean that its too intrusive? Or that its not techincally feasible?

The reason that I ask is that main benefit of collecting analytics is that it allows us to measure daily, weekly, and monthly unique users which we might not be able to do if we don't collect IP addresses.

Reply:
I feel it's too intrusive.
In the case of Gradio-lite, this telemetry will be sent from clients' device immediately after they open the app page, where they don't have any chance to set `analytics_enabled=false` to opt-out it. I think some amount of people consider it as malware that automatically accesses the third-party API (`https://checkip.amazonaws.com/`) to get the IP address and send it as analytics data, under recent situations where people are sensitive to data privacy.

Reply:
I pushed a commit including the IP address fetcher (43af9a3), though we should revert it if it looks not good anyway.